src/OpenLoco/src/Windows/ObjectSelectionWindow.cpp

```diff
@@ -286,12 +313,41 @@ namespace OpenLoco::Ui::Windows::ObjectSelectionWindow
             != a.end();
     }
 
+    static std::optional<VehicleType> getVehicleTypeFromObject(TabObjectEntry& entry)
```

Comment:
Even with the cache, this is proving to be rather expensive (in the order of seconds). Ideally, this information would be available in the object cache...